pkg/compiler/compiler.go

```diff
@@ -27,6 +29,7 @@ type (
 	ProviderPlugin interface {
 		Plugin
 		Translate(result *core.ConstructGraph, dag *core.ResourceGraph) ([]core.CloudResourceLink, error)
+		Validate(config *config.Application, constructGraph *core.ConstructGraph) error
```

Comment:
Rather than all plugins having `Validate`, what about creating a new interface for just this method, and then checking in `compiler.go` whether it's of that type? That way, the default "no validation needed" (like in the k8s plugin" becomes just "don't write the method at all".

Reply:
i think that would cause the import cycle which i was fighting with was the issue otherwise i fully agree. I was hoping to put that method on the provider interface itself, but then we cant type check it in the compiler. Thoughts?

Reply:
It shouldn't cause any import cycle, because the structs don't need to (and in fact can't) import this interface to implement it. I think all you'd need to do is, here do:

```go
ValidatingPlugin interface
	Validate(config *config.Application, constructGraph *core.ConstructGraph) error
}
```

and then below, where you have:

```
err := p.Validate(c.Document.Configuration, c.Document.Constructs)
if err != nil { ...
```

... instead do:

```go
if validator, ok := p.(ValidatingPlugin); ok {
	err := validator.Validate(c.Document.Configuration, c.Document.Constructs)
	if err != nil { ...
}
```

and then just rm the `Kubernetes.Validate` method.

Just to be clear, this isn't a blocker.

Reply:
right just seeing if theres a more optimal way. i think i understand what you are mentioning and may like that better so could give it a try